ui/app/src/services/content.ts

```diff
@@ -564,17 +565,26 @@ function insertCollectionItem(
   }
 }
 
-export function uploadDataUrl(site: string, file: any, path: string, xsrfArgumentName: string): Observable<any> {
+function createFileUpload(
+  uploadUrl: string,
+  file: any,
+  path: string,
+  metaData: object,
+  xsrfArgumentName: string
+): Observable<StandardAction> {
   const qs = toQueryString({ [xsrfArgumentName]: getRequestForgeryToken() });
   return new Observable((subscriber) => {
     const uppy = Core({ autoProceed: true });
-    const uploadAssetUrl = `/studio/asset-upload${qs}`;
-    uppy.use(XHRUpload, { endpoint: uploadAssetUrl });
-    uppy.setMeta({ site, path });
+    uppy.use(XHRUpload, { endpoint: `${uploadUrl}${qs}` });
+    uppy.setMeta(metaData);
 
     const blob = dataUriToBlob(file.dataUrl);
 
-    uppy.on('upload-success', () => {
+    uppy.on('upload-success', (file, response) => {
+      subscriber.next({
+        type: 'upload-success',
+        payload: response.body
```

Comment:
What's this response? Is this the XHR's response object or the API response? We should be returning the whole API response object.